dask_ml/datasets.py

```diff
@@ -381,10 +381,11 @@ def make_classification(
     return X, y
 
 
-def random_date(start, end):
+def random_date(start, end, random_state=None):
+    rng_random_date = dask_ml.utils.check_random_state(random_state)
```

Comment:
Nit:

```suggestion
    rng_random_date = sklearn.utils.check_random_state(random_state)
```

That way the `.compute()` can be avoided (especially relevant on repeated calls.).